### PR TITLE
Add a slew of authenticated features

### PR DESCRIPTION
**Pull request by stevenlafl**
Added some authenticated functionality:

getUserProfileLoggedIn
getThreadsLoggedIn
mute
unmute
block
unblock
getNotifications
setNotificationsSeen
search
getRecommended

To test it I've been running it here in the `newapi` branch: https://github.com/stevenlafl/threads-web-client/tree/newapi

It seems to mostly work, though I'm for some reason having trouble with mute/unmute.

### CODE REVIEW

File: threads-api/src/threads-api.ts [hunk 1]
```diff
@@ -934,9 +1070,229 @@ export class ThreadsAPI {
...
+    let data = {
+      _uid: this.userID,
+      _uuid: this.deviceID,
+      container_module: 'ig_text_feed_timeline'
+    } as {
+      media_id?: string;
+      _uid: string;
+      _uuid: string;
+      container_module: string;
+      target_posts_author_id?: string;
```
Comment (aleclarson) [inline threads-api/src/threads-api.ts:1086]: `target_posts_author_id` is required even when a post ID is given

File: threads-api/src/threads-api.ts [hunk 2]
```diff
@@ -934,9 +1070,229 @@ export class ThreadsAPI {
...
+  getNotifications: PaginationNotificationsQuerier<GetNotificationsPaginatedResponse> = async (
+    filter,
+    pagination,
+    options = {},
+  ): Promise<GetNotificationsPaginatedResponse> => {
+
+    let params: GetNotificationsOptions = {
+			feed_type: 'all',
+			mark_as_seen: false,
+			timezone_offset: -25200,
```
Comment (junhoyeo) [inline threads-api/src/threads-api.ts:1194]: Can we have the `timezone_offset` from the user locale, as in #50?

https://github.com/junhoyeo/threads-api/blob/200ad48d33dc8afe6d14a6e627ac36b1fe711820/threads-api/src/threads-api.ts#L1328-L1333

We should manage this value as an memoized getter...

**Comment by stevenlafl** [comment]
I can't dedicate time right now to this unfortunately, I have a lot coming up in the next two weeks at work and some travel. Please feel free to contribute to it and then merge. If you need to me to convert from Draft to an actual PR let me know.

**Comment by junhoyeo** [comment]
> I can't dedicate time right now to this unfortunately, I have a lot coming up in the next two weeks at work and some travel. Please feel free to contribute to it and then merge. If you need to me to convert from Draft to an actual PR let me know.

Gotcha, will do! Thanks @stevenlafl ❤️

File: threads-api/src/threads-api.ts [hunk 3]
```diff
@@ -1006,7 +1147,7 @@ export class ThreadsAPI {
     if (this.verbose) {
       console.debug('[FOLLOW]', res.data);
     }
-    return res.data;
+    return res.data.status === 'ok';
```
Comment (junhoyeo) [inline threads-api/src/threads-api.ts:1150]: Reverting those to return `res.data` — the `friendship` response can really come in handy :)

**Review by junhoyeo — APPROVED** [review]

**Pull request merged**